oak-search/src/test/java/org/apache/jackrabbit/oak/plugins/index/ExcerptTest.java

```diff
@@ -0,0 +1,340 @@
+
+public abstract class ExcerptTest extends AbstractQueryTest {
+
+    protected IndexOptions indexOptions;
+    protected TestRepository repositoryOptionsUtil;
+
+    protected void assertEventually(Runnable r) {
+        TestUtils.assertEventually(r,
+                ((repositoryOptionsUtil.isAsync() ? repositoryOptionsUtil.defaultAsyncIndexingTimeInSeconds : 0) + 3000) * 5);
+    }
+
+    @Before
+    public void setup() throws Exception { //named so that it gets called after super.before :-/
+        Tree rootTree = root.getTree("/");
+
+        Tree def = rootTree.addChild(INDEX_DEFINITIONS_NAME).addChild("testExcerpt");
+        def.setProperty(JcrConstants.JCR_PRIMARYTYPE, INDEX_DEFINITIONS_NODE_TYPE, Type.NAME);
+        def.setProperty(TYPE_PROPERTY_NAME, indexOptions.getIndexType());
+        def.setProperty(REINDEX_PROPERTY_NAME, true);
+        def.setProperty(FulltextIndexConstants.EVALUATE_PATH_RESTRICTION, true);
+        def.setProperty(FulltextIndexConstants.COMPAT_MODE, IndexFormatVersion.V2.getVersion());
+
+        Tree properties = def.addChild(FulltextIndexConstants.INDEX_RULES)
+                .addChild("nt:base")
+                .addChild(FulltextIndexConstants.PROP_NODE);
+
+        Tree notIndexedProp = properties.addChild("baz");
+        notIndexedProp.setProperty(FulltextIndexConstants.PROP_NODE_SCOPE_INDEX, true);
+
+        Tree relativeProp = properties.addChild("relative-baz");
+        relativeProp.setProperty(FulltextIndexConstants.PROP_ANALYZED, true);
+        relativeProp.setProperty(FulltextIndexConstants.PROP_USE_IN_EXCERPT, true);
+        relativeProp.setProperty(FulltextIndexConstants.PROP_NAME, "relative/baz");
+
+        Tree allProps = properties.addChild("allProps");
+        allProps.setProperty(FulltextIndexConstants.PROP_ANALYZED, true);
+        allProps.setProperty(FulltextIndexConstants.PROP_NODE_SCOPE_INDEX, true);
+        allProps.setProperty(FulltextIndexConstants.PROP_USE_IN_EXCERPT, true);
+        allProps.setProperty(FulltextIndexConstants.PROP_NAME, FulltextIndexConstants.REGEX_ALL_PROPS);
+        allProps.setProperty(FulltextIndexConstants.PROP_IS_REGEX, true);
+
+        root.commit();
+    }
+
+    @Test
+    public void getAllSelectedColumns() throws Exception {
+        Tree contentRoot = root.getTree("/").addChild("testRoot");
+        contentRoot.setProperty("foo", "is fox ifoxing");
+        contentRoot.setProperty("bar", "ifoxing fox");
+        contentRoot.setProperty("baz", "fox ifoxing");
+        root.commit();
+
+        List<String> columns = newArrayList("rep:excerpt", "rep:excerpt(.)", "rep:excerpt(foo)", "rep:excerpt(bar)");
+        String selectColumns = Joiner.on(",").join(
+                columns.stream().map(col -> "[" + col + "]").collect(Collectors.toList())
+        );
+        assertEventually(() -> {
+            String query = "SELECT " + selectColumns + " FROM [nt:base] WHERE CONTAINS(*, 'fox')";
+            Result result = null;
```

Comment:
null init is redundant

```suggestion
            Result result;
```